packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -441,19 +459,59 @@ export class SDKClient {
       Hbar.fromTinybars(Math.floor(networkGasPriceInTinyBars * constants.MAX_GAS_PER_SEC)),
     );
 
-    return {
-      fileId,
-      txResponse: await this.executeTransaction(
+    let txResponse;
+    try {
+      txResponse = await this.executeTransaction(
         ethereumTransaction,
         callerName,
         interactingEntity,
         requestDetails,
         true,
         originalCallerAddress,
-      ),
+      );
+    } catch (e: any) {
+      if (e instanceof SDKClientError && (e.isConnectionDropped() || e.isTimeoutExceeded())) {
+        if (!(await this.isFailedTransaction(requestDetails, e.transactionId))) {
+          txResponse = { transactionId: e.transactionId };
+        } else {
+          throw e;
+        }
+      }
```

Comment:
This block doesn't sseem sufficient enough for handling errors. This [block here](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/lib/clients/sdkClient.ts#L723-L725) also throw a JsonRpcError, and this block doesn't catch that. Maybe the `else` on line 476 need to be 1 level out.

Reply:
The ideas is to not catch all errors and scenarios, here we only want to check the MN for the transaction if we get a timeout exceeded, becuase that when a transaction may have been submitted but the SDK dropped the connection and we didnt get a response. Thats the issue descirbed by Nana on mainnet, we have a timeout exceeded but the trasnaction can be found on hashscan. The other types of errors are handled here https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/lib/eth.ts#L1482

Reply:
This still doesn’t seem to handle `JsonRpcError` correctly. If the error thrown back is an instance of `SDKClientError`, it will remain in this catch block and won’t be thrown again.